ironfish/src/network/peers/peerConnectionManager.ts

```diff
@@ -106,9 +111,43 @@ export class PeerConnectionManager {
       }
     }
 
+    this.maintainMaxPeerCount()
+
     this.eventLoopTimer = setTimeout(() => this.eventLoop(), EVENT_LOOP_MS)
   }
 
+  /**
+   * Maintain a maximum number of peers by disconnecting from peers if we are
+   * connected to more than we should be
+   */
+  private maintainMaxPeerCount(): void {
+    const connectedPeers = this.peerManager.getConnectedPeers()
+    const maxPeerCount = this.maxPeers - Number(this.keepOpenPeerSlot)
```

Comment:
is this like controlling whether this feature is on or not? if so, would it be better to just add an if statement and exit the function if the flag is false? More readable that way?

Reply:
> is this controlling whether this feature is on or not?

No, this explicitly adds the behavior to shed peers if above maxPeers regardless of the flag being enabled. This is helpful in combination with #4350 where we open the door of being above maxPeers. This line just determines whether we're going to shed if peers are _above_ maxPeers (default behavior), or if we are _at_ maxPeers (to maintain an open slot, if this flag is enabled.